src/sql/src/kafka_util.rs

```diff
@@ -246,6 +250,132 @@ pub async fn test_config(
     Ok(())
 }
 
+/// Returns start offsets for the partitions of `topic` and the provided
+/// `time_offset` option.
+///
+/// For each partition, the returned offset is the earliest offset whose
+/// timestamp is greater than or equal to the given timestamp for the
+/// partition. If no such message exists (or the Kafka broker is before
+/// 0.10.0), the current end offset is returned for the partition.
+///
+/// The provided `time_offset` option must be a non-zero number:
+/// * Non-Negative numbers will used as is (e.g. `1622659034343`)
+/// * Negative numbers will be translated to a timestamp in millis
+///   before now (e.g. `-10` means 10 millis ago)
+///
+/// If `time_offset` has not been configured, an empty Option is returned.
+pub async fn lookup_start_offsets(
```

Comment:
We could merge this with the existing `test_config` to avoid another `fetch_metadata` call, but I left it as is to avoid premature optimization and keep the concerns separate.

Reply:
Thanks! I'm reusing the client from the `test_config` fn now (renamed to `create_consumer` which still seems reasonable for the existing doc comment).